lib/core/src/Cardano/Wallet/Primitive/Types.hs

```diff
@@ -1870,3 +1889,49 @@ invariant msg a predicate =
 distance :: (Ord a, Num a) => a -> a -> a
 distance a b =
     if a < b then b - a else a - b
+
+data PoolMetadataSource
+    = None
+    | FetchDirect
+    | FetchSMASH URI
+    deriving (Show, Generic, Eq)
+
+instance ToText PoolMetadataSource where
+    toText None = (T.pack "")
+    toText FetchDirect = (T.pack "direct")
+    toText (FetchSMASH uri) = T.pack $ uriToString id uri ""
+
+instance FromText PoolMetadataSource where
+    fromText "" = Right None
+    fromText "direct" = Right FetchDirect
+    fromText (T.unpack -> uri) =
+        case parseURI uri of
+            Nothing -> Left $ TextDecodingError ("Could not parse URI: " <> uri)
+            Just uri' -> Right (FetchSMASH uri')
+
+data Settings = Settings {
```

Comment:
HLint is gonna bother you to make this a newtype :shrug:.

Reply:
Any reason we want different strictness properties?

Reply:
> Screw hlint :)

That's always an option. If you end up doing this, use a proper pragma to ignore the hlint suggestion with a comment explaining your rationale and why it is justified.

Reply:
I don't see a justification for the rule in the first place. newtype is for:

1. you really want to make sure the data type is strict and zero-cost over the inner type
2. it is morally a newtype (e.g. we want to add instances over an existing type, etc)

Just because a product type currently has only one record doesn't mean you want to turn it into a newtype.

Reply:
I do fully agree with you @hasufell :) 
Perhaps a better way forward about that one would be to disable the rule globally in our hlint config.